statev2/src/applicator/mod.rs

```diff
@@ -66,8 +66,8 @@ impl StateApplicator {
             StateTransition::AddWallet { wallet } => self.add_wallet(&wallet),
             StateTransition::UpdateWallet { wallet } => self.update_wallet(&wallet),
             StateTransition::AddOrder { order } => self.new_order(order),
-            StateTransition::AddOrderValidityProof { order_id, proof } => {
-                self.add_order_validity_proof(order_id, proof)
+            StateTransition::AddOrderValidityProof { order_id, proof, witness } => {
+                self.add_order_validity_proof(order_id, proof, witness)
```

Comment:
out of curiosity, why do we need to share witness w/ peers?
at any rate, why is this an `Option` as opposed to being necessary?

Reply:
we share the witness with cluster peers so that they may use it when performing matches. It holds important information like the reblinded shares of the wallet, or the augmented shares for a given order. It's optional because we _also_ have validity proofs from _other_ clusters. They of course will not share their witness